### ISSUE TITLE
React Checkbox doesn't correctly set checked on double-mount

### ISSUE THREAD

**Issue by karlvr** [OnsenUI/OnsenUI#2418]
__Environment__

Onsen UI Version:

- 2.9.2

Framework:

- React 16.3

Framework binding:

- react-onsen 1.10.0

Additional libraries:

- 

Platform:

- macOS Chrome

__Encountered problem__

Due to some other issue, the `Checkbox` component appears to mount twice. The first time it mounts the `componentWillReceiveProps` method in `BaseInput` is called, and the `checked` property of the `<input>` is set. However the component is mounted again (still trying to work out how, as this doesn't happen in my playground) and the `<input>` is reset to not being checked. This is resolved by setting `checked` in the `componentDidMount`.

This has lead me to observing that `componentWillReceiveProps` seems to be deprecated (https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops) and that perhaps the issue I'm seeing will be resolved by converting to using the `componentDidUpdate` method, which _seems_ to be the successor life cycle method for what we're trying to do.

__How to reproduce__

I have not yet got a playground to reproduce this issue. It seems to be caused by a double-render which I am attempting to reproduce and get to the bottom of.

**Comment by karlvr** [comment]
My work in progress playground:

  
  [__Demo link__](https://onsen.io/playground/?issue)

  

- __HTML__

```html
<div id="app"></div>
```

- __JS__

```jsx
var MyPage = React.createClass({
  
  renderToolbar: function(route, navigator) {
    return (
      <Ons.Toolbar>
        <div className='center'>{route.title}</div>
      </Ons.Toolbar>
    );
  },
  
  dataSource: function() {
    return ['a', 'b', 'c'];
  },
  
  renderRow: function(row, index) {
    return (
      <Ons.ListItem key={index}>
        <label className="left">
          <Ons.Checkbox checked={true} />
        </label>
        <label className="center">
          {row}
        </label>
      </Ons.ListItem>
    );
  },
  
  pushPage: function(navigator) {
    navigator.pushPage({
      title: 'Another page'
    });
  },
  
  renderPage: function(route, navigator) {
    if (route.title === 'First page') {
      return (
        <Ons.Page key={route.title} renderToolbar={this.renderToolbar.bind(this, route, navigator)}>
          <Ons.Button onClick={this.pushPage.bind(this, navigator)}>
            Push Page
          </Ons.Button>
        </Ons.Page>
      );
    } else if (route.title === 'Another page') {
      return (
        <Ons.Page key={route.title} renderToolbar={this.renderToolbar.bind(this, route, navigator)}>
          <Ons.List dataSource={this.dataSource()} renderRow={this.renderRow} />
        </Ons.Page>
      );
    }
  },

  render: function() {
    return (
      <Ons.Navigator renderPage={this.renderPage} initialRoute={{title: 'First page'}} />
    );
  }
});

ons.ready(function() {
  ReactDOM.render(<MyPage />, document.getElementById('app'));
});
```

**Comment by karlvr** [comment]
I've discovered where the second `componentDidMount` is coming from. Using an integration between `RouterNavigator` and `Redux` (which I put together, based on some other examples, https://github.com/karlvr/onsenui-react-redux-navigator) there is a Redux action dispatched on `postPush` that causes the Redux state to change, and thus the components to re-mount.

It's a little dissatisfactory that the Redux integration causes this double-mount, but it feels like it's not breaking React rules, so I _think_ this is something that could be fixed in Onsen's `BaseInput`. As above, I have tested a fix. I'm happy to provide a PR so we can review and discuss.
